tests/ragger/test_domain_name.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+from ragger.error import ExceptionRAPDU
+from app.client import EthereumClient, SettingType
+import struct
+
+# Values used across all tests
+CHAIN_ID = 1
+NAME = "ledger.eth"
+LONG_NAME = "ledger-LONG" + "1234567890"*24 + ".eth"
+ADDR = bytes.fromhex("0011223344556677889900112233445566778899")
+KEY_ID = 1
+ALGO_ID = 1
+BIP32_PATH = [ 44, 60, 0, None, None ]
+NONCE = 21
+GAS_PRICE = 13000000000
+GAS_LIMIT = 21000
+AMOUNT = 1.22
+
+@pytest.fixture(params=[False, True])
+def verbose(request) -> bool:
+    return request.param
+
+def common(app_client: EthereumClient) -> int:
+    if app_client._client.firmware.device == "nanos":
+        pytest.skip("Not supported on LNS")
+    return app_client.get_challenge()
+
+
+def test_send_fund(app_client: EthereumClient, verbose: bool):
+    challenge = common(app_client)
+
+    if verbose:
+        app_client.settings_set({
+            SettingType.VERBOSE_ENS: True
+        })
+
+    app_client.provide_domain_name(challenge, NAME, ADDR)
+
+    app_client.send_fund(BIP32_PATH,
+                         NONCE,
+                         GAS_PRICE,
+                         GAS_LIMIT,
+                         ADDR,
+                         AMOUNT,
+                         CHAIN_ID,
+                         "domain_name_" + str(verbose))
+
+def test_send_fund_wrong_challenge(app_client: EthereumClient):
+    caught = False
+    challenge = common(app_client)
+
+    try:
+        app_client.provide_domain_name(~challenge & 0xffffffff, NAME, ADDR)
+    except ExceptionRAPDU:
+        caught = True
+
+    assert caught == True
```

Comment:
We have a dedicated API in ragger to tweak the exception handling, example from Exchange :
```
app_client.backend.raise_policy = RaisePolicy.RAISE_NOTHING
rapdu: RAPDU = app_client.provide_domain_name(~challenge & 0xffffffff, NAME, ADDR)
assert rapdu.status == 0xABCD
```